leshan-core/src/main/java/org/eclipse/leshan/senml/SenMLPackSerDes.java

```diff
@@ -126,4 +131,64 @@ public SenMLPack deserialize(JsonArray array) {
 
         return pack;
     }
+
+    public byte[] serializeToCbor(SenMLPack pack) {
+        CBORFactory factory = new CBORFactory();
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+        try {
+            CBORGenerator generator = factory.createGenerator(out);
+            generator.writeStartArray();
+
+            for (SenMLRecord record : pack.getRecords()) {
+                generator.writeStartObject();
+
+                if (record.getBaseName() != null && record.getBaseName().length() > 0) {
```

Comment:
Why not using `!record.getBaseName().isEmpty()` ?

Reply:
Not all records have base name within same pack, so that only generate base name element when record contains base name and length great than 0.

Reply:
I meant replacing `record.getBaseName().length() > 0` by `!record.getBaseName().isEmpty()`
(keep the  `record.getBaseName() != null`)